spi/storage/storage.go

```diff
@@ -116,11 +116,21 @@ type Tag struct {
 	Value string `json:"value,omitempty"`
 }
 
+// PutOptions represents options for a Put Operation.
+type PutOptions struct {
+	// This is an optimization for a Put Operation. Some storage providers may be able to store data faster if they
+	// know beforehand that this key does not currently exist in the database. Unexpected behaviour may occur if
+	// this is set to true and the key already exists. See the documentation for the specific storage provider to
+	// see if and how this option is used.
+	IsNewKey bool `json:"isNewKey,omitempty"`
+}
+
 // Operation represents an operation to be performed in the Batch method.
 type Operation struct {
-	Key   string `json:"key,omitempty"`
-	Value []byte `json:"value,omitempty"` // A nil value will result in a delete operation.
-	Tags  []Tag  `json:"tags,omitempty"`  // Optional.
+	Key        string     `json:"key,omitempty"`
+	Value      []byte     `json:"value,omitempty"`      // A nil value will result in a delete operation.
+	Tags       []Tag      `json:"tags,omitempty"`       // Optional.
+	PutOptions PutOptions `json:"putOptions,omitempty"` // Optional. Only used for Put Operations.
```

Comment:
I wonder if it would be better to use a pointer here?

Reply:
Yes - done!